dev/core#6021 SearchKit - Allow form filters to target specific join

colemanw opened the pull request:
Overview
----------------------------------------
Makes it possible to apply a filter to the conditions of a join instead of the whole search. For Optional (LEFT) joins, this means the row will not be completely filtered out from the search, only the joined columns. 

<img width="2122" height="1130" alt="image" src="https://github.com/user-attachments/assets/7e4a0162-65ee-46bc-b3c9-dc8a3065ff51" />

colemanw committed:
Afform - Load join labels server-side

colemanw committed:
Afform - Update afGuiField: modernize JS

colemanw committed:
SearchKit - Allow form filters to target specific join

Fixes dev/core#6021

larssandergreen commented:
@colemanw Thanks so much for doing this, will be super helpful to have.

1. I think from a user perspective, it might be a little more complex and confusing that it needs to be. What if instead of getting all the joins for the search, it just checked if the field in question was on an entity in a left join? Then you could do something like:
```[ ] Filter only Contribution```
instead of the select. It only makes sense to filter within the join if it is for the relevant entity and it is a left join anyways.

2. Testing with a text field using auto operator produces no results. This is because it is producing SQL with CONTAINS instead of LIKE, so you get a syntax error.

colemanw committed:
SearchKit - Fix filters passed to ON clause

Because ON and WHERE clauses behave differently in Api4, this adds normalization,
so that passing FALSE for the 4th param of any clause will always treat the 3rd
param as a literal value, even in the ON clause.

colemanw commented:
@larssandergreen I was able to reproduce the bug you described and have added a fix + unit test for it. Marking this as merge-ready.
